### PR TITLE
dev-env: pin msys2/mingw

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
Today the postgresql 12.4 artifacts were purged from the msys2 primary mirrors. I was able to find a copy on some old mirrors (I downloaded from [mirrors.piconets.webwerks.in], [mirrors.aliyun.com], and [mirror.iscas.ac.cn]), but that raised the question of where to put the binary so our CI nodes can access it. I decided to put it on one of our GCS buckets, and while I was there I did the whole msys2 set of packages.

Does this manually may not be the best path forward for a long-term solution (though maybe it is; they tend not to change all that often), but it does unblock us right now.

For the postgresql binary specifically, I have checked that the three mirrors give me the exact same file between them, and that it matches the old hash. For the other files, I downloaded them from the primary mirror ([repo.msys2.org]) and checked that the hashes match the recorded values in our JSON manifest.

CHANGELOG_BEGIN
CHANGELOG_END

[mirrors.piconets.webwerks.in]: https://mirrors.piconets.webwerks.in/msys2-mirror/mingw/x86_64/mingw-w64-x86_64-postgresql-12.4-1-any.pkg.tar.zst
[mirrors.aliyun.com]: https://mirrors.aliyun.com/msys2/mingw/x86_64/mingw-w64-x86_64-postgresql-12.4-1-any.pkg.tar.zst?spm=a2c6h.25603864.0.0.32b23240Qp8jCL
[mirror.iscas.ac.cn]: https://mirror.iscas.ac.cn/msys2/mingw/x86_64/mingw-w64-x86_64-postgresql-12.4-1-any.pkg.tar.zst
[repo.msys2.org]: https://repo.msys2.org/mingw/x86_64/

### CODE REVIEW

**Comment by garyverhaegen-da** [comment]
Note: our Windows CI nodes are currently busted because of the missing PostgreSQL binary in the primary mirrors. They will need to be reset after this gets merged to main.

File: dev-env/windows/manifests/msys2.json
Comment (stefanobaghino-da) [inline dev-env/windows/manifests/msys2.json:4]: Is this here as a backup or does it have a meaning?

**Review by stefanobaghino-da — APPROVED** [review]

File: dev-env/windows/manifests/msys2.json
Comment (garyverhaegen-da) [reply dev-env/windows/manifests/msys2.json:4]: This is ignored by everything currently. I thought it would be useful to keep track of where the artifacts were originally found. This is not a great long-term solution as there's nothing keeping that in sync with the "local copies" in the GCS bucket, but I thought designing a good long-term solution could wait until we have a working CI again.

Comment (stefanobaghino-da) [reply dev-env/windows/manifests/msys2.json:4]: Probably worth adding a more descriptive string that `_url`, just in case someone wonders about the meaning and you're not around.

**Pull request merged**
